eng/common/pipelines/templates/steps/verify-changelog.yml

```diff
@@ -0,0 +1,44 @@
+parameters:
+  PackageName: 'not-specified'
+  ServiceName: 'not-specified'
+  ForRelease: ''
+
+steps:
+  - pwsh: |
+      Import-Module "$(Build.SourcesDirectory)/eng/common/scripts/modules/common-manifest.psd1"
+      $Language = "$(Build.Repository.Name)".Substring("$(Build.Repository.Name)".LastIndexOf('-') + 1)
+      $PackageProp = Get-PkgProperties -PackageName ${{ parameters.PackageName }} -ServiceName ${{ parameters.ServiceName }} -Language $Language -RepoRoot $(Build.SourcesDirectory)
+
+      $ReleaseNotes = (& "$(Build.SourcesDirectory)/eng/common/Extract-ReleaseNotes.ps1" -ChangeLogLocation $PackageProp.pkgChangeLogPath -VersionString $PackageProp.pkgVersion -IncludeTitle $true) -Split [Environment]::NewLine
+
+      # Verify Header Has Unreleased or an has accurate date
+      $ReleaseTitle = $ReleaseNotes[0] -Split ' '
+
+      if ($ReleaseTitle.Length -ne 3)
+      {
+        Write-Host ("##[error]Changelog '{0}' has wrong release note title" -f $PackageProp.pkgChangeLogPath)
+        Write-Host "##[info]Ensure the release date is included i.e. (yyyy-MM-dd) or (Unreleased) if not yet released"
+        exit 1
+      }
+
+      if ('${{ parameters.ForRelease }}' -eq 'true')
+      {
+        
+        $CurrentDate = Get-Date -Format "yyyy-MM-dd"
+        if ($ReleaseTitle[2] -ne "($CurrentDate)")
```

Comment:
More logic that should be in our change log scripts. It seems to me like this might be a good opportunity to consolidate the Extract and Update Change log scripts to have a common set of functions (either in a common script or module) for working with and validating change log entries. We shouldn't add another script like this that makes assumptions about the structure of our changelogs.

Reply:
I am not sure how this is going to work out. Right now the various languages consume the Update-Change-Log.ps1 script in different ways. [JS](https://github.com/Azure/azure-sdk-for-js/search?q=Update-Change-Log&unscoped_q=Update-Change-Log), [Python](https://github.com/Azure/azure-sdk-for-python/search?q=Update-Change-Log&unscoped_q=Update-Change-Log), [Java](https://github.com/Azure/azure-sdk-for-java/search?q=Update-Change-Log&unscoped_q=Update-Change-Log), .[Net](https://github.com/Azure/azure-sdk-for-net/search?q=Update-Change-Log&unscoped_q=Update-Change-Log) I am not sure how to get it all working if I refactor it into a function ina single module. It will work for DotNet but not sure about the other languages.